core/src/test/kotlin/com/strumenta/kolasu/parsing/KolasuParserTest.kt

```diff
@@ -51,7 +51,7 @@ class KolasuParserTest {
             """.trimMargin()
         )
         assertNotNull(result)
-        val lexingResult = parser.tokenFactory.extractTokens(result)
+        val lexingResult = (parser.tokenFactory as ANTLRTokenFactory).extractTokens(result)
```

Comment:
Should the `extractTokens` method be exposes by TokenFactory to avoid this cast?

Reply:
No because it used to be exposed by `TokenFactory` but internally it assumed ANTLR tokens which is incorrect. That's just a test, user code is not supposed to do that.